### PR TITLE
chore(auth): Allow other sign in types from AuthStateMachine

### PR DESCRIPTION
**Pull request by tylerjroach**
- [ ] PR title and description conform to [Pull Request](https://github.com/aws-amplify/amplify-android/blob/main/CONTRIBUTING.md#pull-request-guidelines) guidelines.

*Issue #, if available:*

*Description of changes:*
Prepping for hosted ui sign in and other sign in methods

*How did you test these changes?*
(Please add a line here how the changes were tested)

- [ ] Added Unit Tests
- [ ] Tested on frontend (Add Screenshots)
- [ ] Successful logs (Attach them to the PR)

*Documentation update required?*
- [ ] No
- [ ] Yes (Please include a PR link for the documentation update)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by gpanshu — DISMISSED** [review]

File: aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/AuthConfiguration.kt [hunk 1]
```diff
@@ -20,77 +20,30 @@ import org.json.JSONObject
...
-        init {
-            configJson?.run {
-                optJSONObject(Config.COGNITO_USER_POOL.key)?.getJSONObject(configName)?.let {
-                    userPool = UserPoolConfiguration.fromJson(it).build()
-                }
-
-                optJSONObject(Config.CREDENTIALS_PROVIDER.key)?.getJSONObject(
-                    Config.COGNITO_IDENTITY.key
-                )
+            configName: String = "Default"
```
Comment (div5yesh) [inline aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/AuthConfiguration.kt:35]: We would want to support multiple configs, such as prod, beta, etc. Should we still have enum to represent this?

File: aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/states/HostedUISignInState.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package com.amplifyframework.statemachine.codegen.states
```
Comment (div5yesh) [inline aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/states/HostedUISignInState.kt:1]: Missing license header.

File: aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/AuthConfiguration.kt [hunk 3]
```diff
@@ -20,77 +20,30 @@ import org.json.JSONObject
...
+        ): AuthConfiguration {
+            return AuthConfiguration(
+                userPool = pluginJson.optJSONObject("CognitoUserPool")?.getJSONObject(configName)?.let {
```
Comment (div5yesh) [inline aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/AuthConfiguration.kt:38]: This is support multiple providers. Let's discuss this.

**Review by div5yesh — COMMENTED** [review]
Few comments.

File: aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/AuthConfiguration.kt (hunk 3 above)
Comment (tylerjroach) [reply aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/AuthConfiguration.kt:38]: The enum was ok, but it seemed to be a bit too generic. Ex: a config enum class with config strings for different areas. An enum should have a bit more focus.

`enum class AuthEnv { DEFAULT, PROD, DEV }`

I'd propose we keep as is until we make the change to support different config names.

File: aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/states/HostedUISignInState.kt (hunk 2 above)
Comment (tylerjroach) [reply aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/states/HostedUISignInState.kt:16]: pushed

**Review by div5yesh — APPROVED** [review]

**Review by gpanshu — APPROVED** [review]

**Comment by tylerjroach** [comment]
Closing. This is a hosted UI PR that is no longer relevant.

**Pull request closed without merging**